napari/layers/shapes/shapes.py

```diff
@@ -2588,7 +2591,9 @@ def remove_selected(self):
             )
         self.selected_data = set()
         self._finish_drawing()
-        self.events.data(value=self.data)
+        self.events.data(
+            value=self.data, action=ActionType.REMOVE.value, data_indices=index
+        )
```

Comment:
why here is no `vertex_indices`?

Reply:
Because the whole shape(s) is removed. Same with adding a shape I do not emit `vertex_indices` there. Only when the `shape` is changed by a move or vertex insert / remove event.

Reply:
But then user needs to add 

```
if hasattr(event, "vertex_indices"):
   ...
```
This is not a user-friendly interface.

Reply:
Would one not first check the action and the layer source in any case?

Reply:
I could imagine that someone passes it as arguments to some function when the action check is done. 

In general, the interface should be always the same. Maybe you should add an empty sequence, but the `vertex_indices` should be defined.

Reply:
resolved with [d0ce75a](https://github.com/napari/napari/pull/5967/commits/d0ce75a3f19dc41fb6bbb54d6bfb98b7b657a046)